src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Controller/DefaultControllerTest.php

```diff
@@ -27,30 +27,48 @@ protected function setUp()
     /**
      * @param $getValueMap
      * @param $uri
+     *
      * @return \PHPUnit_Framework_MockObject_MockObject
      */
-    private function getRequestMock($uri, $url, $redirect = null)
+    private function getRequestMock($requestUrl, $portalUrl, $redirectUrl = null)
     {
         $request = $this->getMockBuilder('Symfony\Component\HttpFoundation\Request')->getMock();
         $request->expects($this->any())->method('get')->will(
             $this->returnValueMap(
                 array(
-                    array('url', null, false, $url),
-                    array('redirect', null, false, $redirect)
+                    array('url', null, false, $portalUrl),
+                    array('redirect', null, false, $redirectUrl)
                 )
             )
         );
-        $request->expects($this->any())->method('getUri')->will($this->returnValue($uri));
+        $request->expects($this->any())->method('getUri')->will($this->returnValue($requestUrl));
 
         return $request;
     }
 
-    public function testRedirectActionWithTrailingSlash()
+    public function provideRedirectAction()
     {
-        $request = $this->getRequestMock('sulu-redirect.lo/', 'sulu-redirect.lo', 'sulu.lo');
+        return array(
+            array('http://sulu.lo/articles?foo=bar', 'sulu.lo', 'sulu.lo/en', 'http://sulu.lo/en/articles?foo=bar'),
+            array('http://sulu.lo/articles?foo=bar&bar=boo', 'sulu.lo', 'sulu.lo/en', 'http://sulu.lo/en/articles?foo=bar&bar=boo'),
+            array('sulu.lo:8001/', 'sulu.lo', 'sulu.lo/en', 'http://sulu.lo:8001/en'),
+            array('sulu.lo:8001/#foobar', 'sulu.lo', 'sulu.lo/en', 'http://sulu.lo:8001/en/#foobar'),
```

Comment:
I guess that's what @alexander-schranz meant, the result here should be `http://sulu.lo:8001/en/#foobar`